client/cli/src/commands/export_blocks_cmd.rs

```diff
@@ -14,39 +14,39 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-use std::io;
-use std::fs;
-use std::path::PathBuf;
-use std::fmt::Debug;
+use crate::error;
+use crate::params::{BlockNumber, PruningParams, SharedParams};
+use crate::CliConfiguration;
 use log::info;
-use structopt::StructOpt;
 use sc_service::{
-	Configuration, ServiceBuilderCommand, ChainSpec,
-	config::DatabaseConfig, Roles,
+	config::DatabaseConfig, Configuration, ServiceBuilderCommand,
 };
 use sp_runtime::traits::{Block as BlockT, Header as HeaderT};
-
-use crate::error;
-use crate::VersionInfo;
-use crate::runtime::run_until_exit;
-use crate::params::{SharedParams, BlockNumber, PruningParams};
+use std::fmt::Debug;
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+use structopt::StructOpt;
 
 /// The `export-blocks` command used to export blocks.
 #[derive(Debug, StructOpt, Clone)]
 pub struct ExportBlocksCmd {
 	/// Output file name or stdout if unspecified.
+	// NOTE: this is an option so implementations can set their own defaults
 	#[structopt(parse(from_os_str))]
 	pub output: Option<PathBuf>,
 
 	/// Specify starting block number.
 	///
 	/// Default is 1.
+	// NOTE: this is an option so implementations can set their own defaults
 	#[structopt(long = "from", value_name = "BLOCK")]
 	pub from: Option<BlockNumber>,
 
 	/// Specify last block number.
 	///
 	/// Default is best block.
+	// NOTE: this is an option so implementations can set their own defaults
```

Comment:
wait no. why am I removing this. This has been requested by @gnunicorn . Can you check together and let met know if I should keep them or not?

Reply:
These are just code comments(aka not seen by anyone) and they don't bring any value. If the value is none, the implementation needs to provide a default value or ignore it 🤷‍♂️

Reply:
The comment you showed me from @gnunicorn is about showing default values in the cli. Which is not easy doable with the current implementation. But this comment does not bring any value.

Reply:
The comment she is referring to was in a personal conversation. I was making the point that why this is being an `Option` that always has a default is non-obvious and easily something that someone in the future on a refactor or otherwise touching this code might consider optimizing and thus breaking it without realising (and that is something that is also easily overlooked by a reviewer). However that is future relevant, yet undocumented at this point, and I don't want to have this regression because of simple oversight. So I proposed to put in a comment that makes clear to a future developer, why this is an `Option`. As we have no other safeguard to on this, I propose keeping the comment – it is an easy and quick safeguard.

Reply:
Reiterating from a conversation here – note that I care only 3 out of 10 about this. I don't think it is obvious from the code that having an `Option` is an explicitly wanted feature and thus the likelihood of this being refactored away unknowingly is pretty high. Vigilance on the side of the reviewer can prevent that from happening, sure, and the comment increases the chances of that happening. Another easy recommendation to ensure this feature is not being removed accidentally would also be to have a test case `test_that_XYZ_default_can_be_provided_by_implementation` rather than the comment. This would make it even more clear why this is the way it is. 

However, I do not insist on either of these.  If you believe vigilance from the reviewer is fine, alright with me, too. You are owners of this code ;) .

Reply:
I don't have opinion on this tbh. It seems to me that the comment can easily be missed but it's not wrong to have one. Making a test is probably better but I'm not entirely sure it will be clear, someone can just update the test without seeing the point. So I will do whatever you guys decide.

The reason why it is here is because we use structopt in a library and it's not really made for that imo... Time will tell if what we choose here was the right decision or not.